### PR TITLE
feat: Method that create a title with a given view 🍕 

### PR DESCRIPTION
**Pull request by a7maad-ayman**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature

## Description
Create a function that generate header section with a provided UIView and String title

## Related Tickets & Documents
Home<>Method that create a title with a given view

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: Healthy.xcodeproj/project.pbxproj
Comment (ahmdyasser) [inline Healthy.xcodeproj/project.pbxproj:1]: Remove the empty lines.

File: Healthy/Classes/Modules/Dashboard/DashboardViewController.swift [hunk 1]
```diff
@@ -33,6 +33,54 @@ final class DashboardViewController: UIViewController {
 
         configureView()
     }
+
+    // MARK: - Function for header section creation
```
Comment (ahmdyasser) [inline Healthy/Classes/Modules/Dashboard/DashboardViewController.swift]: If you want to describe what the function does, don't use **// MARK**, instead use documentation comment. You can do so by pressing `options + cmd + /`

image


Those info will show when you get the definition of a function like this:

image

Also this type of comment will help you to generate documentation for your app using DocC.

File: Healthy/Classes/Modules/Dashboard/DashboardViewController.swift [hunk 2]
```diff
@@ -33,6 +33,54 @@ final class DashboardViewController: UIViewController {
...
+
+        NSLayoutConstraint.activate([
+            content.leadingAnchor.constraint(equalTo: titleLabel.trailingAnchor, constant: 10),
+            content.topAnchor.constraint(equalTo: stack.topAnchor),
+            content.trailingAnchor.constraint(equalTo: stack.trailingAnchor, constant: -8),
+            content.bottomAnchor.constraint(equalTo: content.topAnchor, constant: -8)
+        ])
+
+        return stack
+    }
```
Comment (ahmdyasser) [inline Healthy/Classes/Modules/Dashboard/DashboardViewController.swift]: It's a good practice to select all and indent the code whenever you finish writing code in a file, or you can download SwiftFormat to do the formatting for you.

```suggestion
    func makeHeaderSection(title: String, content: UIView) -> UIView {
        let titleLabel = UILabel()
        let stack = UIStackView(arrangedSubviews: [titleLabel, content])

        stack.translatesAutoresizingMaskIntoConstraints = false
        titleLabel.translatesAutoresizingMaskIntoConstraints = false
        content.translatesAutoresizingMaskIntoConstraints = false

        titleLabel.text = title
        view.addSubview(stack)
        stack.backgroundColor = .darkGray
        stack.axis = .horizontal
        stack.alignment = .center
        stack.distribution = .fillEqually
        stack.spacing = 10

        NSLayoutConstraint.activate([
            stack.leadingAnchor.constraint(
                equalTo: view.safeAreaLayoutGuide.leadingAnchor,
                constant: 8),
            stack.trailingAnchor.constraint(
                equalTo: view.safeAreaLayoutGuide.trailingAnchor,
                constant: -8),
            stack.topAnchor.constraint(
                equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 8),
            stack.heightAnchor.constraint(equalToConstant: view.bounds.height*0.1)
        ])

        NSLayoutConstraint.activate([
            titleLabel.leadingAnchor.constraint(equalTo: stack.leadingAnchor, constant: 8),
            titleLabel.trailingAnchor.constraint(equalTo: content.leadingAnchor),
            titleLabel.topAnchor.constraint(equalTo: stack.topAnchor),
            titleLabel.bottomAnchor.constraint(equalTo: content.topAnchor, constant: -8)
        ])

        NSLayoutConstraint.activate([
            content.leadingAnchor.constraint(equalTo: titleLabel.trailingAnchor, constant: 10),
            content.topAnchor.constraint(equalTo: stack.topAnchor),
            content.trailingAnchor.constraint(equalTo: stack.trailingAnchor, constant: -8),
            content.bottomAnchor.constraint(equalTo: content.topAnchor, constant: -8)
        ])

        return stack
    }
}
```

Comment (ahmdmhasn) [inline Healthy/Classes/Modules/Dashboard/DashboardViewController.swift]: This is how is think of it... No constraints are needed. 

```suggestion
// MARK: - Headers Creation

private extension DashboardViewController {
    private func makeSection(title: String, content: UIView) -> UIView {

        let titleLabel = UILabel()
        titleLabel.text = title
        titleLabel.applyStyle(TitleLabelStyle())

        let stackView = UIStackView(arrangedSubviews: [titleLabel, content])
        stackView.axis = .horizontal
        stackView.alignment = .fill
        stackView.spacing = 8.0

        return stackView
    }
}
```

This is an example usage 
```swift
    func configureView() {
        stackView.addArrangedSubview(headerView)

        // Section with header example
        stackView.addArrangedSubview(
            makeSection(
                title: "Header",
                content: UIView()
            )
        )
    }
```

Comment (ahmdmhasn) [reply Healthy/Classes/Modules/Dashboard/DashboardViewController.swift]: Please don't forget to add one unit test case at least and add a screenshot of the result in the description.

File: Healthy.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -14,8 +14,10 @@
 		022924302A33A170009290A8 /* UIColors.Generated.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0229242D2A33A170009290A8 /* UIColors.Generated.swift */; };
 		0252BF192A365A6F00823339 /* CreateAccountViewModelType.swift in Sources */ = {isa = PBXBuildFile; fileRef = 0252BF182A365A6F00823339 /* CreateAccountViewModelType.swift */; };
 		025511922A3C394300295B91 /* Colors.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 20376AA82A2663F8004BF0BF /* Colors.xcassets */; };
+
```
Comment (ahmdmhasn) [inline Healthy.xcodeproj/project.pbxproj]:
```suggestion
```

File: Healthy.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -14,8 +14,10 @@
...
 		025511962A3C3F6400295B91 /* Poppins-Regular.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 025511942A3C3F6400295B91 /* Poppins-Regular.ttf */; };
 		025511972A3C3F6400295B91 /* Poppins-Bold.ttf in Resources */ = {isa = PBXBuildFile; fileRef = 025511952A3C3F6400295B91 /* Poppins-Bold.ttf */; };
+
```
Comment (ahmdmhasn) [inline Healthy.xcodeproj/project.pbxproj]:
```suggestion
```

**Comment by ahmdmhasn** [comment]
@a7maad-ayman Any update?

**Comment by a7maad-ayman** [comment]
> @a7maad-ayman Any update?

**Comment by a7maad-ayman** [comment]
> > @a7maad-ayman Any update?

I have made the requested changes in this PR :
 - Rebasing the `develop` branch into the local `feat/header-section` branch to incorporate the latest changes.
 - Removed unnecessary spaces and `MARK` comments .
 - Refactored the method into a new extension without any constraints 
 
 #####  Please note that I did not include Unit Tests specifically for the private method as it is not directly accessible for testing.

**Review by ahmdmhasn — APPROVED** [review]

**Pull request merged**